rollouts/controllers/rollout_controller.go

```diff
@@ -448,18 +451,22 @@ func (r *RolloutReconciler) computeTargets(ctx context.Context,
 	return targets, nil
 }
 
-func pkgNeedsUpdate(ctx context.Context, rollout *gitopsv1alpha1.Rollout, rs gitopsv1alpha1.RemoteSync, pkg *packagediscovery.DiscoveredPackage) (*gitopsv1alpha1.RemoteSync, bool) {
-	// TODO: We need to check other things here besides git.Revision and metadata
-	metadata := getSpecMetadata(rollout)
-	if pkg.Revision != rs.Spec.Template.Spec.Git.Revision ||
-		!reflect.DeepEqual(metadata, rs.Spec.Template.Metadata) ||
-		rs.Spec.Type != rollout.GetSyncTemplateType() {
+// rsNeedsUpdate checks if the underlying remotesync needs to be updated by creating a new RemoteSync object and comparing it to the existing one
+func rsNeedsUpdate(ctx context.Context, rollout *gitopsv1alpha1.Rollout, oldRS *gitopsv1alpha1.RemoteSync, target *clusterPackagePair) (*gitopsv1alpha1.RemoteSync, bool) {
+	newRS := newRemoteSync(rollout,
+		gitopsv1alpha1.ClusterRef{Name: target.cluster.Name},
+		toSyncSpec(target.packageRef),
+		pkgID(target.packageRef),
+		getSpecMetadata(rollout),
+	)
 
-		rs.Spec.Template.Spec.Git.Revision = pkg.Revision
-		rs.Spec.Template.Metadata = metadata
-		rs.Spec.Type = rollout.GetSyncTemplateType()
-		return &rs, true
+	// if the spec of the new RemoteSync object is not identical to the existing one, then an update is necessary
+	if !equality.Semantic.DeepEqual(oldRS.Spec, newRS.Spec) {
+		oldRS.Spec = newRS.Spec
```

Comment:
Some suggestions:
Renaming `oldRS` to `currRS` and `newRS` to `expRS` or `desiredRS` will make the code even more obvious.

Reply:
Will do